File: apps/keke_driver/lib/features/driver/domain/driver_profile.dart
enum DriverStatus {
  unregistered,
  pendingDocuments,
  pendingApproval,
  approved,
  suspended,
  rejected
}

enum OperationStatus {
  offline,
  available,
  busy
}

enum TripStep {
  none,
  accepted,
  arrived,
  started,
  completed
}

class DriverProfile {
  final String? id;
  final String? firstName;
  final String? lastName;
  final DriverStatus status;
  final String? vehiclePlate;
  final String? vehicleModel;
  final String? licenseUrl;
  final String? idCardUrl;
  final String? vehiclePaperUrl;
  final String? photoUrl;
  final double debtAmount;
  final bool ninVerified;

  const DriverProfile({
    this.id,
    this.firstName,
    this.lastName,
    required this.status,
    this.vehiclePlate,
    this.vehicleModel,
    this.licenseUrl,
    this.idCardUrl,
    this.vehiclePaperUrl,
    this.photoUrl,
    this.debtAmount = 0.0,
    this.ninVerified = false,
  });

  DriverProfile copyWith({
    String? id,
    String? firstName,
    String? lastName,
    DriverStatus? status,
    String? vehiclePlate,
    String? vehicleModel,
    String? licenseUrl,
    String? idCardUrl,
    String? vehiclePaperUrl,
    String? photoUrl,
    double? debtAmount,
    bool? ninVerified,
  }) {
    return DriverProfile(
      id: id ?? this.id,
      firstName: firstName ?? this.firstName,
      lastName: lastName ?? this.lastName,
      status: status ?? this.status,
      vehiclePlate: vehiclePlate ?? this.vehiclePlate,
      vehicleModel: vehicleModel ?? this.vehicleModel,
      licenseUrl: licenseUrl ?? this.licenseUrl,
      idCardUrl: idCardUrl ?? this.idCardUrl,
      vehiclePaperUrl: vehiclePaperUrl ?? this.vehiclePaperUrl,
      photoUrl: photoUrl ?? this.photoUrl,
      debtAmount: debtAmount ?? this.debtAmount,
      ninVerified: ninVerified ?? this.ninVerified,
    );
  }
}

File: apps/keke_backend/src/models/DriverProfile.ts
import { Entity, PrimaryColumn, Column, CreateDateColumn, UpdateDateColumn, Index } from "typeorm";

export enum DriverStatus {
    PENDING_DOCUMENTS = "pending_documents",
    PENDING_REVIEW = "pending_review",
    APPROVED = "approved",
    REJECTED = "rejected",
    SUSPENDED = "suspended"
}

@Entity()
export class DriverProfile {
    @Index()
    @PrimaryColumn()
    userId!: string; // References User.id

    @Column()
    firstName!: string;

    @Column()
    lastName!: string;

    @Column()
    vehiclePlate!: string;

    @Column()
    vehicleModel!: string;

    @Index()
    @Column({ type: "enum", enum: DriverStatus, default: DriverStatus.PENDING_DOCUMENTS })
    status!: DriverStatus;

    @Column({ type: 'varchar', length: 500, nullable: true })
    rejectionReason!: string;

    @Column({ nullable: true })
    licenseUrl!: string;

    @Column({ nullable: true })
    idCardUrl!: string;

    @Column({ nullable: true })
    vehiclePaperUrl!: string;

    @Column({ nullable: true })
    photoUrl!: string;

    @Column({ type: 'varchar', length: 50, nullable: true, default: null })
    nin!: string | null;

    @Column({ default: false })
    ninVerified!: boolean;

    /**
     * Denormalized passenger-rating aggregates. Average = ratingSum / ratingCount
     * (0 when ratingCount is 0). Stored as sum+count rather than a float average
     * so each new review is an exact, race-safe increment.
     */
    @Column({ type: "int", default: 0 })
    ratingSum!: number;

    @Column({ type: "int", default: 0 })
    ratingCount!: number;

    // ── Park operations (Phase 2) ───────────────────────────────────────

    /**
     * What device this driver actually has.
     *
     * The reason Park Dispatch exists: a large share of Keke drivers own no
     * Android phone, or own one and keep mobile data off. Assuming a smartphone
     * is how a system ends up serving only the drivers who least need it.
     *
     * Defaults to SMARTPHONE because every existing driver signed up through
     * the app — but nothing may INFER capability from that default; it is a
     * declared property, corrected by whoever onboards the driver.
     */
    @Column({ type: "varchar", length: 20, default: "smartphone" })
    deviceCapability!: "smartphone" | "feature_phone" | "none";

    /**
     * The number painted on the Keke. What a dispatcher reads across a park,
     * and the largest element on a printed badge.
     */
    @Index()
    @Column({ type: "varchar", length: 24, nullable: true })
    unitNumber!: string | null;

    /** The park this driver normally works out of. Roster membership is separate. */
    @Index()
    @Column({ type: "varchar", nullable: true })
    homeParkId!: string | null;

    @CreateDateColumn()
    createdAt!: Date;

    @UpdateDateColumn()
    updatedAt!: Date;
}
